packages/content-conformance/src/rules/ensure-valid-nav-paths.js

```diff
@@ -0,0 +1,71 @@
+import flat from 'flat'
+
+// Recursively traverse json nav data, and validate node paths against flat list of files from context
+function visitor(navData, filePaths, report) {
+  // flatten nav data file
+  const flatNodes = flat(navData)
+  // get all the paths from the nav data file
+  const navDataPaths = Object.entries(flatNodes)
+    .filter(([key]) => {
+      return key.endsWith('.path')
+    })
+    .map(([, value]) => value)
+
+  navDataPaths.forEach((p) => {
+    const namedFile = `${p}.mdx`
+    const indexFile = `${p}/index.mdx`
+
+    const namedFileExists = filePaths.includes(namedFile)
+    const indexFileExists = filePaths.includes(indexFile)
+
+    if (!namedFileExists && !indexFileExists) {
+      report(
+        `NavData path (${p}) is missing a corresponding file in the content directory`
+      )
+    }
+
+    if (namedFileExists && indexFileExists) {
+      report(
+        `NavData path (${p}) should not have both a named file and an index file in the content directory`
+      )
+    }
+  })
+}
+
+/** @type {import('../types.js').ConformanceRuleBase} */
+export default {
+  id: 'ensure-valid-nav-paths',
+  type: 'data',
+  description: 'TODO.',
+  executor: {
+    async dataFile(file, context) {
+      // by convention, nav data files are those that end in nav-data.json
+      if (!file.basename.endsWith('nav-data.json')) {
+        return
+      }
+
+      const subpath = file.basename.split('-nav-data.json')[0]
```

Comment:
@BRKalow we could have this internal mapping...

```suggestion
      let subpath = file.basename.split('-nav-data.json')[0]
      // handle small variances in our implicit nav-data-to-folder mapping
      subpath =
        {
          'plugin-framework': 'plugin/framework',
          'plugin-sdk': 'plugin/sdkv2',
          'plugin-log': 'plugin/log',
          'plugin-mux': 'plugin/mux',
          'cloud-docs-agents': 'cloud-docs/agents',
        }[subpath] || subpath
```

... or potentially avoid this and expose a mapping config option (?)

Reply:
@thiskevinwang I'm surprised we don't have this mapping elsewhere already 🤔 Ideally, it would be preferable if we could handle this automatically. Could we check if a path segment exists that matches `subpath`? For example, if I the `subpath` is `api-docs`, we would have a path segment that matches 1:1. For `plugin-framework`, there would be no match so we would instead check for `plugin/framework`.

Reply:
Ok this is what I arrived on to avoid a mapping object 🤔 - https://github.com/hashicorp/web-platform-packages/pull/137/commits/68600b297d5ad258c32b6843e840187a1b1208bf